File: src/commands/status.ts
import chalk from "chalk";
import { loadAuth, loadConfig, loadState, expandPath } from "../config/store";

export async function statusCommand(): Promise<void> {
	const auth = await loadAuth();
	const config = await loadConfig();
	const state = await loadState();

	console.log();

	// Connection status
	if (auth?.deviceToken) {
		console.log(`  ${chalk.green("●")} Connected`);
	} else {
		console.log(`  ${chalk.red("●")} Not connected`);
		console.log(`  Run ${chalk.cyan("remarkable-cli init")} to connect.`);
		console.log();
		return;
	}

	// Sync folder
	console.log(`  Sync folder: ${chalk.cyan(expandPath(config.syncFolder))}`);

	// Last sync
	if (state.lastSyncTimestamp > 0) {
		const date = new Date(state.lastSyncTimestamp);
		const relative = getRelativeTime(date);
		console.log(`  Last sync:   ${date.toLocaleString()} (${relative})`);
	} else {
		console.log(`  Last sync:   ${chalk.dim("never")}`);
	}

	// Item counts
	const docs = Object.values(state.syncState).filter(r => r.type === "DocumentType").length;
	const collections = Object.values(state.syncState).filter(r => r.type === "CollectionType").length;
	if (docs > 0 || collections > 0) {
		console.log(`  Documents:   ${docs}`);
		console.log(`  Folders:     ${collections}`);
	}

	// Last report
	if (state.lastSyncReport) {
		const r = state.lastSyncReport;
		const duration = r.durationMs < 1000 ? `${r.durationMs}ms` : `${(r.durationMs / 1000).toFixed(1)}s`;
		console.log();
		console.log(chalk.dim(`  Last sync: ${r.syncedCount} synced, ${r.skippedCount} unchanged, ${r.failedCount} failed (${duration})`));
	}

	// Config
	if (config.excludePdfs) {
		console.log(chalk.dim(`  PDFs excluded`));
	}

	console.log();
}

function getRelativeTime(date: Date): string {
	const now = Date.now();
	const diff = now - date.getTime();
	const minutes = Math.floor(diff / 60_000);
	const hours = Math.floor(diff / 3_600_000);
	const days = Math.floor(diff / 86_400_000);

	if (minutes < 1) return "just now";
	if (minutes < 60) return `${minutes}m ago`;
	if (hours < 24) return `${hours}h ago`;
	return `${days}d ago`;
}

File: src/commands/status.rs
use crate::config::{ConfigStore, OutputFormat};
use crate::error::Result;
use colored::Colorize;
use std::time::{SystemTime, UNIX_EPOCH};

pub async fn run(store: &ConfigStore) -> Result<()> {
    let auth = store.load_auth().await?;
    let config = store.load_config().await?;
    let state = store.load_state().await?;
    let sync_root = store.expand_home(&config.sync_folder)?;

    println!();
    if auth.is_registered() {
        println!("{} Registered", "●".green());
    } else {
        println!("{} Not registered", "●".red());
        println!("Run `remarkable init` to connect.");
        println!();
        return Ok(());
    }

    println!("Sync folder: {}", sync_root.display().to_string().cyan());
    println!(
        "Output format: {}",
        match config.output_format {
            OutputFormat::Pdf => "pdf",
            OutputFormat::Markdown => "markdown",
        }
    );

    match state.last_sync_timestamp {
        Some(timestamp) => println!("Last sync: {}", relative_time(timestamp)),
        None => println!("Last sync: never"),
    }

    let document_count = state
        .records
        .values()
        .filter(|record| record.item_type == "DocumentType")
        .count();
    let collection_count = state
        .records
        .values()
        .filter(|record| record.item_type == "CollectionType")
        .count();
    println!("Documents: {document_count}");
    println!("Folders: {collection_count}");

    if let Some(report) = state.last_sync_report {
        println!(
            "Last report: {} synced, {} skipped, {} failed",
            report.synced_count, report.skipped_count, report.failed_count
        );
    }
    println!();
    Ok(())
}

fn relative_time(timestamp: u64) -> String {
    let now = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .unwrap_or_default()
        .as_secs();
    let diff = now.saturating_sub(timestamp);

    if diff < 60 {
        "just now".to_string()
    } else if diff < 3_600 {
        format!("{}m ago", diff / 60)
    } else if diff < 86_400 {
        format!("{}h ago", diff / 3_600)
    } else {
        format!("{}d ago", diff / 86_400)
    }
}
